awscli/clidocs.py

```diff
@@ -173,6 +173,8 @@ def doc_option(self, arg_name, help_command, **kwargs):
             argument.argument_model, argument.cli_type_name)))
         doc.style.indent()
         doc.include_doc_string(argument.documentation)
+        if getattr(argument.argument_model, 'is_tagged_union', False):
```

Comment:
One thing that I noticed is that it would be great if we can update the type in the docs similar to what we did for streaming blobs (e.g. make the type `(tagged union structure)`) next to the field name. I'm mainly suggesting this because it will help reconfirm what field the note is for.